main.go

```diff
@@ -143,6 +146,24 @@ func prepare() *config.Config {
 
 func main() {
 	relayConfig := prepare()
+	if *cpuprofile != "" {
+		f, err := os.Create(*cpuprofile)
+		if err != nil {
+			log.Fatal(err)
+		}
+		pprof.StartCPUProfile(f)
+		defer pprof.StopCPUProfile()
+	}
+	if *memprofile != "" {
+		defer func() {
+			f, err := os.Create(*memprofile)
```

Comment:
Would it be better to create the profile file when Relay starts (and thus fail immediately if the file can't be created) rather than doing it all in a deferred function?